From 62cb8c5de15f2a6ee08d27f4fe713df04e7052c0 Mon Sep 17 00:00:00 2001
From: Ghennadi Procopciuc <ghennadi.procopciuc@nxp.com>
Date: Wed, 23 Mar 2022 08:57:46 +0200
Subject: [PATCH 044/103] arch: s32-cc: Remove get_effective_memsize

s32-cc version of get_effective_memsize isn't needed as
the default one can successfully initialize the size of DDR.

Issue: ALB-8576
Upstream-Status: Pending 

Signed-off-by: Ghennadi Procopciuc <ghennadi.procopciuc@nxp.com>
Signed-off-by: Zhantao Tang <zhantao.tang@windriver.com>
---
 arch/arm/mach-s32/s32-cc/cpu.c | 30 ------------------------------
 1 file changed, 30 deletions(-)

diff --git a/arch/arm/mach-s32/s32-cc/cpu.c b/arch/arm/mach-s32/s32-cc/cpu.c
index 4017596133..9af173dfd9 100644
--- a/arch/arm/mach-s32/s32-cc/cpu.c
+++ b/arch/arm/mach-s32/s32-cc/cpu.c
@@ -125,33 +125,3 @@ int arch_cpu_init(void)
 	return 0;
 }
 
-phys_size_t __weak get_effective_memsize(void)
-{
-	unsigned long size;
-	int nodeoff = -1, ret;
-	struct fdt_resource res = {.start = 0, .end = 0};
-
-	/*
-	 * Restrict U-Boot area to the first bank of the DDR memory.
-	 * Note: gd->bd isn't initialized yet
-	 */
-	size = PHYS_SDRAM_1_SIZE;
-
-	/* Get first DDR bank size from DT 'memory' node */
-	while ((nodeoff = fdt_node_offset_by_prop_value(gd->fdt_blob, nodeoff,
-							"device_type",
-							"memory", 7)) >= 0) {
-		ret = fdt_get_resource(gd->fdt_blob, nodeoff, "reg", 0, &res);
-		if (ret) {
-			pr_err("Unable to get 'reg' values of memory node\n");
-			return ret;
-		}
-		if (res.start == PHYS_SDRAM_1) {
-			size = res.end - res.start + 1;
-			break;
-		}
-	}
-
-	return size;
-}
-
-- 
2.25.1

